build/registry/pkg/distribution/index_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package distribution_test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/falcosecurity/plugins/build/registry/pkg/distribution"
+)
+
+const (
+	indexFile         = "testdata/index.yaml"
+	wrongIndexFile    = "testdata/wrong-index.yaml"
+	registryFile      = "testdata/registry.yaml"
+	wrongRegistryFile = "testdata/wrong-registry.yaml"
+	registryUser      = "falcosecurity"
+	registryName      = "ghcr.io"
+)
+
+var _ = Describe("Update index", func() {
```

Comment:
These black box unit test provide the minimum expected behaviour with specifications of the index update from the registry manifest.